### PR TITLE
Start adding a CODEOWNERS file.

### PR DESCRIPTION
**Pull request by Carreau**
I suggest this after a comment from jni, as well as  pr-4932 and 4962
that I might have merged a bit overzealously without waiting for a core
napari contributor review.

I think that having assigned reviewers for part of the codebase,
and enforcing that they review/approve given changes might make ti a bit
clearer who is responsible about what.

Note that this can be made a bit easier with teams. We could have a core
team, doc team, plugins team that could be responsible for the
corresponding folders.

Current this just added Juan to review all PNGs, but that's of course
just an example.

You can find the documentation about CODEOWNERS here for more details.

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

### CODE REVIEW

File: .github/CODEOWNERS
Comment (Carreau) [inline .github/CODEOWNERS]: I invite you to make suggestions if you wish to add yourself to a part of the codebase, like so:

```suggestion
*.png @jni
napari/plugins/ @tlambert03 @nclack 
```

Comment (Czaki) [reply .github/CODEOWNERS]: looking here, more files should be assigned to jni. 
On the other hand, each part should be at least two people not to have a block during someone's holidays.

https://github.com/napari/napari/blob/297dfc3dfc48963181411d64d06199857076ca4c/.gitattributes#L2-L5

Comment (Carreau) [reply .github/CODEOWNERS]: Sure, in the end I expect teams to be assigned, and not single contributors.

**Comment by brisvag** [comment]
Does this simply sned review requests, or also enforce that without those reviews the PR *cannot* be merged?

**Comment by Carreau** [comment]
> Does this simply send review requests, or also enforce that without those reviews the PR _cannot_ be merged?

It depends what is checked in napari repo's settings and what is decided by the core team.

**Comment by jni** [comment]
> Current this just added Juan to review all PNGs

🤣🤣🤣🤣

**Review by jni — APPROVED** [review]
I propose we merge this as-is (thanks for the demo @Carreau! 😂) and then iterate over several PRs. We can discuss which teams we need to form etc in the coming weeks.

**Review by justinelarsen — APPROVED** [review]
This is cool, thanks @Carreau ! I agree let's merge and then create some teams and iterate over the codebase to make sure we're covered.

**Comment by liu-ziyang** [comment]
In the spirit of assigning a group of owners instead of a single person, should we change the png owner from Juan to the steering council group before we merge?

**Comment by Czaki** [comment]
There is a ned for some pr that will contain png. Any idea?

File: .github/CODEOWNERS
Comment (Carreau) [reply .github/CODEOWNERS]:
```suggestion
*.png @napari/core-devs 
*.jpg @napari/core-devs
*.gif @napari/core-devs
*.webm @napari/core-devs
```

**Comment by Carreau** [comment]
> In the spirit of assigning a group of owners instead of a single person, should we change the png owner from Juan to the steering council group before we merge?

Updated to `@napari/core-devs`, and added gif, webm and jpg.

**Comment by Czaki** [comment]
> Updated to `@napari/core-devs`, and added gif, webm and jpg.

but this includes all persons, so it does not change anything from the current state?

**Comment by Carreau** [comment]
> but this includes all persons, so it does not change anything from the current state?

No it includes [only these 15 persons](https://github.com/orgs/napari/teams/core-devs/members), I for example am not part of it, but have commit right

**Comment by Carreau** [comment]
In addition it will request review from the team, which it did not before, but we'll need an owner to go in the setting and enable mandatory review from codewoners for protected branches.

**Review by liu-ziyang — APPROVED** [review]

**Comment by liu-ziyang** [comment]
It seems we have a consensus to merge in this to kickstart the process and iterate on it. Going to merge tomorrow if no objection

**Pull request merged**
